Backend/purrfect-match-backend/src/main/java/com/purrfectmatch/restservice/Breeds/BreedController.java

```diff
@@ -0,0 +1,41 @@
+package com.purrfectmatch.restservice.Breeds;
+
+import com.purrfectmatch.restservice.NegativePageNumberException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin(origins = "*")
+public class BreedController {
+  @Autowired
+  BreedService service;
+
+  private final BreedRepository repository;
+  private Integer pageSize;
+
+  public BreedController(BreedRepository repository) {
+    this.repository = repository;
+    this.pageSize = 10;
+  }
+
+  @PostMapping("/breeds/all")
+  List<Breed> all(@RequestParam(defaultValue = "0") Integer pageNumber) {
+    try {
+      if (pageNumber < 0) {
+        throw new NegativePageNumberException(pageNumber);
+      }
+    } catch (Exception e) {
+      System.out.println("Exiting program...");
+      System.exit(1);
```

Comment:
Oops, this is definitely not what you want. Good for testing, not so good for actual use.